packages/web/src/store/application/ui/withdraw-usdc/sagas.ts

```diff
@@ -177,12 +186,207 @@ function* createDestinationTokenAccount({
+  amount,
+  currentBalance
+}: Pick<
+  ReturnType<typeof beginWithdrawUSDC>['payload'],
+  'amount' | 'currentBalance'
+>) {
+  const { track, make } = yield* getContext('analytics')
+  yield* put(beginCoinflowWithdrawal())
+
+  const analyticsFields: WithdrawUSDCTransferEventFields = {
+    // TODO
+    destinationAddress: 'COINFLOW',
+    amount: amount / 100,
+    // Incoming balance is in cents, analytics values are in dollars
+    currentBalance: currentBalance / 100
+  }
+  try {
+    const audiusBackendInstance = yield* getContext('audiusBackendInstance')
+
+    yield* call(
+      track,
+      make({
+        eventName: Name.WITHDRAW_USDC_REQUESTED,
+        ...analyticsFields
+      })
+    )
+
+    const libs = yield* call(getLibs)
+    if (!libs.solanaWeb3Manager) {
+      throw new Error('Failed to get solana web3 manager')
+    }
+    const rootSolanaAccount = yield* call(getRootSolanaAccount)
+
+    const destinationAddress = rootSolanaAccount.publicKey.toString()
+
+    if (!destinationAddress || !amount) {
+      throw new Error('Please enter a valid destination address and amount')
+    }
+
+    let withdrawalAmount = amount
+    const feePayer = yield* select(getFeePayer)
+    if (feePayer === null) {
+      throw new Error('Missing Fee Payer.')
+    }
+    const feePayerPubkey = new PublicKey(feePayer)
+    const connection = libs.solanaWeb3Manager.connection
+
+    const destinationPubkey = new PublicKey(destinationAddress)
+    let destinationTokenAccountAddress: string
+
+    // Check to see if the address is already an associated token account
+    const isTokenAccountAddress = yield* call(isTokenAccount, {
+      accountAddress: destinationAddress as SolanaWalletAddress,
+      mint: 'usdc'
+    })
+
+    if (isTokenAccountAddress) {
+      // If the destination is already a token account, we can transfer directly
+      destinationTokenAccountAddress = destinationAddress
+    } else {
+      // If it's not, derive the associated token account
+      const destinationWallet = destinationPubkey
+      const destinationTokenAccount = yield* call(
+        getAssociatedTokenAddressSync,
+        libs.solanaWeb3Manager.mints.usdc,
+        destinationWallet
+      )
+      destinationTokenAccountAddress = destinationTokenAccount.toBase58()
+
+      // Ensure the derived token account exists
+      const tokenAccountInfo = yield* call(getTokenAccountInfo, {
+        tokenAccount: destinationTokenAccount,
+        mint: 'usdc'
+      })
+
+      // If not, then create an associated token account
+      if (tokenAccountInfo === null) {
+        console.debug(
+          'Withdraw USDC - destination associated token account does not exist. Creating...'
+        )
+        yield* call(createDestinationTokenAccount, {
+          destinationWallet,
+          destinationTokenAccount,
+          feePayer: feePayerPubkey
+        })
+
+        // At this point, we likely have swapped some USDC for SOL. Make sure that we are able
```

Comment:
Where would we have swapped USDC for sol at this point?

Reply:
my bad, copypasta from the other flow that needs to be cleaned up.

Reply:
Oh wait just kidding. We may have swapped USDC for SOL to fund creating the destination account if it didn't yet exist. That happens in `createDestinationTokenAccount`